### PR TITLE
Create/Update code samples for knative-gcp sources

### LINKED ISSUE #2119: Create/Update code samples for knative-gcp sources.
**Describe the change you'd like to see**
Create/Update code samples for knative-gcp sources. Currently the code samples for GCP Cloud Pub/Sub source is out-dated. And there is no code samples for CloudStorageSource, Cloud SchedulerSource and CloudAuditLogsSource. Need to add code samples for CloudStorageSource, Cloud SchedulerSource and CloudAuditLogsSource. 

**Additional context**

### PR DESCRIPTION
**Pull request by capri-xiyue**
Fixes #2119 

## Proposed Changes

- Create/Update code samples for knative-gcp sources
-
-

### CODE REVIEW

**Comment by capri-xiyue** [comment]
I checked the pull-knative-docs-markdown-link-check failure. It failed because `../../contributing/WORKING-GROUPS.md#eventing doesn't exist`. It is not introduced by this PR. This file `../../contributing/WORKING-GROUPS.md#eventing` is also missing in master branch. Does anyone know whether I should delete this working-groups session or create a new one for it?

**Comment by samodell** [comment]
Hey @capri-xiyue  -- don't worry, the link checker isn't a blocking presubmit. That file actually lives over in the Community repo and is pulled in to the website at build time; you don't need to worry about that particular broken link.

File: docs/eventing/README.md [hunk 1]
```diff
@@ -208,26 +208,65 @@ The GitHubSource fires a new event for selected
...
+
+### CloudStorageSource
+
+Registers for events of the specified types on the specified 
+[Google Cloud Storage](https://cloud.google.com/storage/) bucket and optional object prefix. 
+Brings those events into Knative.
+
+**Spec fields**:
+
+- `bucket`: `string` GCS bucket to subscribe to. For example my-test-bucket.
```
Comment (nachocano) [inline docs/eventing/README.md]: can we add the `project` and `secret` to all of them... not the pubsub secret, because we will eventually remove it

**Comment by nachocano** [comment]
Thank you for updating this @capri-xiyue !!!

File: docs/eventing/README.md [hunk 2]
```diff
@@ -208,26 +208,85 @@ The GitHubSource fires a new event for selected
...
-  A reference to a Secret which contains a GCP refresh token for talking to
-  PubSub.
 - `sink`:
-  [ObjectReference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#objectreference-v1-core)
-  A reference to the object that should receive events.
+  [Destination](https://github.com/knative/pkg/blob/master/apis/duck/v1/destination.go)
+  A reference to the target that should receive events.
+- `secret`:
+  [SecretKeySelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#secretkeyselector-v1-core)
+  Credential to use to poll the Cloud Pub/Sub Subscription. It is not used to create or delete the Subscription, only to poll it.
```
Comment (mattmoor) [inline docs/eventing/README.md]: Remove trailing whitespace:
```suggestion
  Credential to use to poll the Cloud Pub/Sub Subscription. It is not used to create or delete the Subscription, only to poll it.
```

File: docs/eventing/README.md [hunk 3]
```diff
@@ -208,26 +208,85 @@ The GitHubSource fires a new event for selected
...
+  [SecretKeySelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#secretkeyselector-v1-core)
+  Credential to use to poll the Cloud Pub/Sub Subscription. It is not used to create or delete the Subscription, only to poll it. 
+  The value of the secret entry must be a service account key in the JSON format (see https://cloud.google.com/iam/docs/creating-managing-service-account-keys). Defaults to secret.name of 'google-cloud-key' and secret.key of 'key.json'.
+- `project`: `string` ID of the Google Cloud Project that the Pub/Sub Topic exists in. If omitted uses the Project ID from the GKE cluster metadata service.
+
+See the [CloudPubSubSource](samples/cloudpubsubsource) example.
+
+### CloudStorageSource
+
+Registers for events of the specified types on the specified
```
Comment (mattmoor) [inline docs/eventing/README.md]: Remove trailing whitespace:
```suggestion
Registers for events of the specified types on the specified
```

File: docs/eventing/README.md [hunk 4]
```diff
@@ -208,26 +208,85 @@ The GitHubSource fires a new event for selected
...
+  Credential to use to poll the Cloud Pub/Sub Subscription. It is not used to create or delete the Subscription, only to poll it. 
+  The value of the secret entry must be a service account key in the JSON format (see https://cloud.google.com/iam/docs/creating-managing-service-account-keys). Defaults to secret.name of 'google-cloud-key' and secret.key of 'key.json'.
+- `project`: `string` ID of the Google Cloud Project that the Pub/Sub Topic exists in. If omitted uses the Project ID from the GKE cluster metadata service.
+
+See the [CloudPubSubSource](samples/cloudpubsubsource) example.
+
+### CloudStorageSource
+
+Registers for events of the specified types on the specified 
+[Google Cloud Storage](https://cloud.google.com/storage/) bucket and optional object prefix.
```
Comment (mattmoor) [inline docs/eventing/README.md]: Remove trailing whitespace:
```suggestion
[Google Cloud Storage](https://cloud.google.com/storage/) bucket and optional object prefix.
```

File: docs/eventing/README.md [hunk 5]
```diff
@@ -208,26 +208,85 @@ The GitHubSource fires a new event for selected
...
+  [SecretKeySelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#secretkeyselector-v1-core)
+  Credential to use for managing GCS notifications.
+  The value of the secret entry must be a service account key in the JSON format (see https://cloud.google.com/iam/docs/creating-managing-service-account-keys). Defaults to secret.name of 'google-cloud-key' and secret.key of 'key.json'.
+- `project`: `string` ID of the Google Cloud Project that the Pub/Sub Topic exists in. If omitted uses the Project ID from the GKE cluster metadata service.
+
+See the [CloudStorageSource](samples/cloudstoragesource) example.
+
+### CloudSchedulerSource
+
+Create, update, and delete [Google Cloud Scheduler](https://cloud.google.com/scheduler/) Jobs.
```
Comment (mattmoor) [inline docs/eventing/README.md]: Remove trailing whitespace:
```suggestion
Create, update, and delete [Google Cloud Scheduler](https://cloud.google.com/scheduler/) Jobs.
```

File: docs/eventing/README.md [hunk 6]
```diff
@@ -208,26 +208,85 @@ The GitHubSource fires a new event for selected
...
+
+- `serviceName`: `string` The GCP service providing audit logs.
+- `methodName`: `string` The name of the service method or operation. For API calls, this should be the name of the API method.
+- `resourceName`: `string` The resource or collection that is the target of the operation. The name is a scheme-less URI, not including the API service name. If omitted, audit log events matching service and method will be pulled for all resources.
+- `sink`:
+  [Destination](https://github.com/knative/pkg/blob/master/apis/duck/v1/destination.go)
+  A reference to the target that should receive events.
+- `secret`:
+  [SecretKeySelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#secretkeyselector-v1-core)
+  Credential used to pull Stackdriver audit log pubsub messages.
```
Comment (mattmoor) [inline docs/eventing/README.md]: Remove trailing whitespace:
```suggestion
  Credential used to pull Stackdriver audit log pubsub messages.
```

File: docs/eventing/samples/cloudauditlogssource/README.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+---
+title: "CloudAuditLogsSource"
+linkTitle: "CloudAuditLogsSource"
+weight: 20
+type: "docs"
+---
+
+# CloudAuditLogsSource Example
+
+Please refer to the [example](https://github.com/google/knative-gcp/blob/master/docs/examples/cloudauditlogssource/README.md) in knative-gcp.
```
Comment (mattmoor) [inline docs/eventing/samples/cloudauditlogssource/README.md]: Add trailing newline:
```suggestion
Please refer to the [example](https://github.com/google/knative-gcp/blob/master/docs/examples/cloudauditlogssource/README.md) in knative-gcp.

```

File: docs/eventing/samples/cloudpubsubsource/README.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
+---
+title: "CloudPubSubSource"
+linkTitle: "CloudPubSubSource"
+weight: 20
+type: "docs"
+---
+
+# CloudPubSubSource Example
+
+Please refer to the [example](https://github.com/google/knative-gcp/blob/master/docs/examples/cloudpubsubsource/README.md) in knative-gcp.
```
Comment (mattmoor) [inline docs/eventing/samples/cloudpubsubsource/README.md]: Add trailing newline:
```suggestion
Please refer to the [example](https://github.com/google/knative-gcp/blob/master/docs/examples/cloudpubsubsource/README.md) in knative-gcp.

```

File: docs/eventing/samples/cloudschedulersource/README.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+---
+title: "CloudSchedulerSource"
+linkTitle: "CloudSchedulerSource"
+weight: 20
+type: "docs"
+---
+
+# CloudSchedulerSource Example
+
+Please refer to the [example](https://github.com/google/knative-gcp/blob/master/docs/examples/cloudschedulersource/README.md) in knative-gcp.
```
Comment (mattmoor) [inline docs/eventing/samples/cloudschedulersource/README.md]: Add trailing newline:
```suggestion
Please refer to the [example](https://github.com/google/knative-gcp/blob/master/docs/examples/cloudschedulersource/README.md) in knative-gcp.

```

File: docs/eventing/samples/cloudstoragesource/README.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
+---
+title: "CloudStorageSource"
+linkTitle: "CloudStorageSource"
+weight: 20
+type: "docs"
+---
+
+# CloudStorageSource Example
+
+Please refer to the [example](https://github.com/google/knative-gcp/blob/master/docs/examples/cloudstoragesource/README.md) in knative-gcp.
```
Comment (mattmoor) [inline docs/eventing/samples/cloudstoragesource/README.md]: Add trailing newline:
```suggestion
Please refer to the [example](https://github.com/google/knative-gcp/blob/master/docs/examples/cloudstoragesource/README.md) in knative-gcp.

```

**Comment by capri-xiyue** [comment]
@samodell  Hi Sam, could you help review this PR?

**Comment by capri-xiyue** [comment]
assign @samodell

File: docs/eventing/samples/cloudpubsubsource/README.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+---
+title: "CloudPubSubSource"
+linkTitle: "CloudPubSubSource"
+weight: 20
+type: "docs"
```
Comment (samodell) [inline docs/eventing/samples/cloudpubsubsource/README.md]:
```suggestion
type: "docs"
aliases:
  - /docs/eventing/samples/gcp-pubsub-source/README.md
```

Comment (samodell) [reply docs/eventing/samples/cloudpubsubsource/README.md]: This adds a redirect from the deleted file.

File: docs/eventing/samples/cloudauditlogssource/README.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (samodell) [inline docs/eventing/samples/cloudauditlogssource/README.md]: Global: for file names, we should be separating words with hyphens. For example, this should be /samples/cloud-audit-logs-source/ instead.

File: docs/eventing/samples/cloudpubsubsource/README.md [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+---
+
+# CloudPubSubSource Example
```
Comment (nachocano) [inline docs/eventing/samples/cloudpubsubsource/README.md]: from the new install page we have a link to the old gcp-pub-sub-source example. Remember to point it to this one. Thanks!

**Pull request merged**